File: LeetCode/238. Product of Array Except Self/solution.go
func productExceptSelf(nums []int) []int {
	n := len(nums)
	result := make([]int, n)
	left := 1

	for i := 0; i < n; i++ {
		result[i] = left
		left = left * nums[i]
	}
	right := 1
	for i := n - 1; i >= 0; i-- {
		result[i] *= right
		right = right * nums[i]
	}
	return result
}

File: LeetCode/238. Product of Array Except Self/solution.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        res = [1]*n
        left = 1
        for i in range(len(nums)):
            res[i] *= left 
            left *= nums[i]
        
        right = 1
        for i in range(len(nums)-1, -1, -1):
            res[i] *= right
            right *= nums[i]
        return res